modules/account-lib/src/coin/stx/transactionBuilder.ts

```diff
@@ -0,0 +1,267 @@
+import BigNumber from 'bignumber.js';
+import BigNum from 'bn.js';
+import { BaseCoin as CoinConfig } from '@bitgo/statics/dist/src/base';
+import {
+  BufferReader,
+  deserializeTransaction,
+  emptyMessageSignature,
+  isSingleSig,
+  makeSigHashPreSign,
+  publicKeyFromSignature,
+} from '@stacks/transactions';
+import { StacksNetwork, StacksTestnet } from '@stacks/network';
+import { BaseTransactionBuilder } from '../baseCoin';
+import {
+  BuildTransactionError,
+  ParseTransactionError,
+  SigningError,
+  InvalidTransactionError,
+} from '../baseCoin/errors';
+import { BaseAddress, BaseFee, BaseKey } from '../baseCoin/iface';
+import { Transaction } from './transaction';
+import { KeyPair } from './keyPair';
+import { SignatureData } from './iface';
+import { isValidAddress, removeHexPrefix } from './utils';
+
+export abstract class TransactionBuilder extends BaseTransactionBuilder {
+  private _transaction: Transaction;
+  protected _fee: BaseFee;
+  protected _nonce: number;
+  protected _source: BaseAddress;
+  protected _memo: string;
+  protected _numberSignatures: number;
+  protected _multiSignerKeyPairs: KeyPair[];
+  protected _signatures: SignatureData[];
+  protected _network: StacksNetwork;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+    this._multiSignerKeyPairs = [];
+    this._signatures = [];
+    this._numberSignatures = 1;
+    this._network = new StacksTestnet();
+    this.transaction = new Transaction(_coinConfig);
+  }
+
+  /**
+   * Initialize the transaction builder fields using the decoded transaction data
+   *
+   * @param {Transaction} tx the transaction data
+   */
+  initBuilder(tx: Transaction): void {
+    this.transaction = tx;
+    const txData = tx.toJson();
+    this.fee({ fee: txData.fee.toString() });
+    this.source({ address: txData.from });
+    if (txData.payload.memo) {
+      this.memo(txData.payload.memo);
+    }
+    // check if it is signed or unsigned tx
+    if (isSingleSig(tx.stxTransaction.auth.spendingCondition!)) {
+      if (tx.stxTransaction.auth.spendingCondition.signature.data === emptyMessageSignature().data) {
+        // unsigned transaction
```

Comment:
can we remove this dead space? Instead the `else` case should be the default case here

Reply:
something like `if isSingleSig() && data !== emptyMessageSig() {
 ...
}`

Reply:
updated